connectors/gitlab/element-templates/gitlab-connector.json

```
@@ -852,66 +889,562 @@
       "binding": {
         "type": "zeebe:input",
         "name": "url"
       },
       "condition": {
         "property": "operation",
         "oneOf": [
-          "createRelease"
+          "getRepositoryBranchesByProjectId",
+          "createRepositoryBranch"
         ]
       }
     },
     {
       "type": "Hidden",
-      "value": "={\"tag_name\": if releaseTagName = null then null else releaseTagName, \"name\": if releaseName = null then null else releaseName, \"description\": if releaseDescription = null then releaseDescription else releaseDescription, \"ref\": if releaseRef = null then null else releaseRef}",
+      "value": "={\"search\": if searchString = null then null else searchString, \"regex\": if regex = null then null else regex}",
       "binding": {
         "type": "zeebe:input",
-        "name": "body"
+        "name": "queryParameters"
       },
       "condition": {
         "property": "operation",
         "oneOf": [
-          "createRelease"
+          "getRepositoryBranchesByProjectId"
+        ]
+      }
+    },
+    {
+      "label": "Name",
+      "description": "Name of the branch",
+      "group": "operation",
+      "type": "String",
+      "feel": "optional",
+      "binding": {
+        "type": "zeebe:input",
+        "name": "branchName"
+      },
+      "constraints": {
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryBranch"
+        ]
+      }
+    },
+    {
+      "label": "Reference",
+      "description": "Branch name or commit SHA to create branch from",
+      "group": "operation",
+      "type": "String",
+      "feel": "optional",
+      "binding": {
+        "type": "zeebe:input",
+        "name": "ref"
+      },
+      "constraints": {
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryBranch"
+        ]
+      }
+    },
+    {
+      "type": "Hidden",
+      "value": "={\"branch\": if branchName = null then null else branchName, \"ref\": if ref = null then null else ref}",
+      "binding": {
+        "type": "zeebe:input",
+        "name": "body"
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryBranch"
+        ]
+      }
+    },
+    {
+      "label": "Branch name",
+      "description": "Name of the new branch to create the file in. The commit is added to this branch",
+      "group": "operation",
+      "type": "String",
+      "feel": "optional",
+      "binding": {
+        "type": "zeebe:input",
+        "name": "branchName"
+      },
+      "constraints": {
+        "notEmpty": true
+      },
+      "condition": {
+        "property": "operation",
+        "oneOf": [
+          "createRepositoryFile"
+        ]
+      }
+    },
+    {
+      "label": "Commit message",
+      "group": "operation",
```

Consider adding a description here.

---

What description would you imagine here that adds additional context beyond the label?

---

```suggestion
      "description": "Message of the commit that adds the new file",
      "group": "operation",
```

^ How about this? My thinking is: it might be a little confusing to see a "commit' field when you're creating a file.

---

Got it, makes sense! Will add the description.